mycore-webtools/src/main/java/org/mycore/webtools/properties/MCRPropertyHelper.java

```diff
@@ -0,0 +1,124 @@
+import org.mycore.common.config.MCRComponent;
+import org.mycore.common.config.MCRConfigurationDir;
+import org.mycore.common.config.MCRConfigurationInputStream;
+import org.mycore.common.config.MCRRuntimeComponentDetector;
+import org.mycore.common.content.MCRContent;
+import org.mycore.common.content.MCRFileContent;
+import org.mycore.common.content.MCRURLContent;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+class MCRPropertyHelper {
+
+    private final LinkedHashMap<String, byte[]> configFileContents;
+
+    MCRPropertyHelper() throws IOException {
+        configFileContents = getConfigFileContents("mycore.properties");
+    }
+
+    public static LinkedHashMap<String, byte[]> getConfigFileContents(String filename) throws IOException {
+        LinkedHashMap<String, byte[]> map = new LinkedHashMap<>();
+        for (MCRComponent component : MCRRuntimeComponentDetector.getAllComponents()) {
+            try (InputStream is = component.getConfigFileStream(filename)) {
+                if (is != null) {
+                    map.put(component.getName(), is.readAllBytes());
+                }
+            }
+        }
+        // load config file from classpath
+        try (InputStream configStream = getConfigFileStream(filename)) {
+            if (configStream != null) {
+                LogManager.getLogger().debug("Loaded config file from classpath: " + filename);
+                map.put("classpath_" + filename, configStream.readAllBytes());
+            }
+        }
+
+        //load config file from app config dir
+        File localConfigFile = MCRConfigurationDir.getConfigFile(filename);
+        if (localConfigFile != null && localConfigFile.canRead()) {
+            LogManager.getLogger().debug("Loaded config file from config dir: " + filename);
+            try (FileInputStream fis = new FileInputStream(localConfigFile)) {
+                map.put("configdir_" + filename, fis.readAllBytes());
+            }
+        }
+        return map;
+    }
+
+    private static InputStream getConfigFileStream(String filename) throws IOException {
+        File cfgFile = new File(filename);
+        MCRContent input = null;
+        if (cfgFile.canRead()) {
+            input = new MCRFileContent(cfgFile);
+        } else {
+            URL url = MCRConfigurationInputStream.class.getClassLoader().getResource(filename);
+            if (url != null) {
+                input = new MCRURLContent(url);
+            }
+        }
+        return input == null ? null : input.getInputStream();
+    }
+
+    public Map<String, List<MCRProperty>> analyzeProperties() throws IOException {
+        HashMap<String, String> currentProperties = new HashMap<>();
```

Comment:
why not use MCRProperties here? Maybe we could remove interpolatePropertyValue()

Reply:
The `IOException` still has to be propagated because of `properties.load(...)` in line 101.

Reply:
I found another bug and i had to resuse MCRProperties to fix it.